python-client/giskard/ml_worker/core/dataset.py

```diff
@@ -35,7 +35,8 @@ def __init__(
         self.df = pd.DataFrame(df)
         self.target = target
         self.column_types = self.extract_column_types(self.df)
-        self.feature_types = self.extract_feature_types(list(self.column_types.keys()), cat_columns)
+        self.feature_types = feature_types if feature_types else self.extract_feature_types(
+            list(self.column_types.keys()), cat_columns)
```

Comment:
I don't think we should do that. If the user provides us with `cat_columns` that's all the info we need to extract `feature_types`. There's no scenario where we need `feature_types` now that `cat_columns` is implemented. 

P.S. We should change `feature_types` to `variable_types`.